File: court-booking-frontend/packages/types/src/review.ts
export interface Review {
  id: string;
  booking_id: string;
  venue_id: string;
  player_id: string;
  /** Section 32 Part 6: only the player's first name is shown publicly (privacy). */
  player_first_name: string | null;
  rating: number;
  comment: string | null;
  owner_reply: string | null;
  owner_replied_at: string | null;
  /** Section 32 Part 6: admin can hide an abusive review. */
  is_hidden: boolean;
  created_at: string;
  /** Section 32 Part 6: set when the player edits within the 7-day window. */
  updated_at: string | null;
}

export interface CreateReviewInput {
  booking_id: string;
  rating: number;
  comment?: string;
}

export interface UpdateReviewInput {
  rating: number;
  comment?: string;
}

File: court-booking-backend/app/schemas/review.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ReviewCreateIn(BaseModel):
    booking_id: uuid.UUID
    rating: int = Field(ge=1, le=5)
    comment: str | None = None


class ReviewUpdateIn(BaseModel):
    """Player edits their own review within the 7-day window (Section 32 Part 6)."""

    rating: int = Field(ge=1, le=5)
    comment: str | None = None


class ReviewReplyIn(BaseModel):
    owner_reply: str = Field(min_length=1)


class ReviewOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    booking_id: uuid.UUID
    venue_id: uuid.UUID
    player_id: uuid.UUID
    # Only the player's FIRST name is ever shown publicly (privacy). Set by the service.
    player_first_name: str | None = None
    rating: int
    comment: str | None
    owner_reply: str | None
    owner_replied_at: datetime | None
    is_hidden: bool = False
    created_at: datetime
    updated_at: datetime | None = None


class ReviewSummaryOut(BaseModel):
    """Aggregate shown on venue/search cards (Section 32 Part 6)."""

    average_rating: float | None = None
    review_count: int = 0
